pkg/receiver/rawk8seventsreceiver/receiver.go

```diff
@@ -188,6 +262,7 @@ func (r *rawK8sEventsReceiver) processEventChangeLoop() {
 // this includes: checking if we should process the event, converting it into a plog.Logs
 // and sending it to the next consumer in the pipeline
 func (r *rawK8sEventsReceiver) processEventChange(ctx context.Context, eventChange *eventChange) {
+	r.recordEventProcessed(eventChange.event)
```

Comment:
This should only happen after `consumeWithRetry` returns without error. Otherwise we risk losing events in case of errors from the consumer.

Reply:
> Otherwise we risk losing events in case of errors from the consumer.
We do risk losing events in any case. To prevent losing events, we'd need to implement a proper persistent buffer. 

Moving this line to after successful consumption of event is a substitute of a persistent buffer that will work in certain scenarios and will not work in others.

What's more, moving this line to after checking if event is accepted means we are not storing the *latest* resource version received, but rather the *highest* resource version number received. This is some scenarios can lead to data loss where the original solution would lead to data duplication.

Reply:
Maybe change the name to `recordEventReceived`